src/biogeochem/CropType.F90

```diff
@@ -42,12 +42,23 @@ module CropType
      real(r8), pointer :: gddtsoi_patch           (:)   ! patch growing degree-days from planting (top two soil layers) (ddays)
      real(r8), pointer :: vf_patch                (:)   ! patch vernalization factor for cereal
      real(r8), pointer :: cphase_patch            (:)   ! phenology phase (see cphase_* constants above for possible values)
+     integer , pointer :: sowing_reason_patch     (:)   ! reason for most recent sowing of this patch
      real(r8), pointer :: latbaset_patch          (:)   ! Latitude vary baset for hui (degree C)
      character(len=20) :: baset_mapping
      real(r8) :: baset_latvary_intercept
      real(r8) :: baset_latvary_slope
+     integer , pointer :: next_rx_sdate           (:)   ! prescribed sowing date for the next growing season this year
+     integer , pointer :: rx_sdates_thisyr        (:,:) ! all prescribed sowing dates for this patch this year
+     real(r8), pointer :: rx_cultivar_gdds_thisyr (:,:) ! all cultivar GDD targets for this patch this year
      real(r8), pointer :: sdates_thisyr           (:,:) ! all actual sowing dates for this patch this year
+     real(r8), pointer :: sdates_perharv          (:,:) ! all actual sowing dates for crops *harvested* this year
+     real(r8), pointer :: syears_perharv          (:,:) ! all actual sowing years for crops *harvested* this year
      real(r8), pointer :: hdates_thisyr           (:,:) ! all actual harvest dates for this patch this year
+     real(r8), pointer :: gddaccum_thisyr         (:,:) ! accumulated GDD at harvest for this patch this year
+     real(r8), pointer :: hui_thisyr              (:,:) ! accumulated heat unit index at harvest for this patch this year
+     real(r8), pointer :: sowing_reason_thisyr    (:,:) ! reason for each sowing for this patch this year
+     real(r8), pointer :: sowing_reason_perharv   (:,:) ! reason for each sowing of crops *harvested* this year
+     real(r8), pointer :: harvest_reason_thisyr   (:,:) ! reason for each harvest for this patch this year
```

Comment:
I think I did originally have those as integers but ran into compiler issues in the I/O. I'll look into it again and ask for help if needed.

Good idea about using constants for harvest reasons. Would it be possible to save those to output netCDF files as happens for (I think) CFT names?

Reply:
I wondered if that was the reason you used reals... I wouldn't be surprised if there were I/O issues with integers... it might just be that we need to extend one of the I/O routines to handle 2-d integer variables. If you send the error message, I can help look into it, and we can stick with reals if it looks like a non-trivial issue to resolve. In either case, I think using parameters is a good idea (even if they need to be reals rather than integers).

In terms of adding metadata on these parameters: that's a great idea, but it might be less straightforward than for CFT names. Here's the subroutine that does it for CFT names, but it seems like you'll need a different approach here... or actually, I guess you could use a similar approach if you add an array of names describing each of these reasons, which would be analogous to the array of pft/cft names:

https://github.com/ESCOMP/CTSM/blob/616905bbbac55f506fccc5aef8c45bbbdbba94a2/src/main/histFileMod.F90#L2571-L2598

(But it might also be reasonable to just hard-code a string like "1 = Reason X, 2 = Reason Y", etc. in the long name for the variable... I'll leave it up to you how dynamic you want to make this.)

Reply:
I'm going to mentally mark "save things that could be integers as integers" as out of scope for this PR, and instead I'll open a separate issue.